packages/synapse-react-client/src/components/Markdown/MarkdownSynapse.tsx

```diff
@@ -1,14 +1,16 @@
-import SynapseClient from '@/synapse-client'
-import { SynapseClientError, SynapseContext } from '@/utils'
+import {
```

Comment:
Strategy for migrating this was basically:

1. Extract utility functions and move to MarkdownUtils
2. Extract standalone components and pull out of MarkdownSynapse (still in this file)
3. Retrieve data from Synapse API react-query hooks
4. Add `useEffect`s to replace effectful code that ran after retrieving data or on component props/state change.

The logic for rendering markdown has not changed. Some bugs have likely been fixed by converting the effectful code to `useEffect`s with clear dependencies.